services/wallet/datastore.go

```diff
@@ -560,51 +552,26 @@ var (
 	ErrGeoResetDifferent = errors.New("geo reset is different")
 )
 
-// LinkWallet links a wallet together
-func (pg *Postgres) LinkWallet(ctx context.Context, ID string, userDepositDestination string, providerLinkingID uuid.UUID, anonymousAddress *uuid.UUID, depositProvider, country string) error {
-	sublogger := logger(ctx).With().Str("wallet_id", ID).Logger()
-	sublogger.Debug().Msg("linking wallet")
-
-	// rep check
-	if repClient, ok := ctx.Value(appctx.ReputationClientCTXKey).(reputation.Client); ok {
-		walletID, err := uuid.FromString(ID)
-		if err != nil {
-			sublogger.Warn().Err(err).Msg("invalid wallet id")
-			return fmt.Errorf("invalid wallet id, not uuid: %w", err)
-		}
-		// we have a client, check the value for ID
-		reputable, cohorts, err := repClient.IsLinkingReputable(ctx, walletID, country)
-		if err != nil {
-			sublogger.Warn().Err(err).Msg("failed to check reputation")
-			return fmt.Errorf("failed to check wallet rep: %w", err)
-		}
+// LinkWallet links a rewards wallet to the given deposit provider.
+func (pg *Postgres) LinkWallet(ctx context.Context, id string, userDepositDestination string, providerLinkingID uuid.UUID, depositProvider, country string) error {
+	walletID, err := uuid.FromString(id)
+	if err != nil {
+		return fmt.Errorf("invalid wallet id, not uuid: %w", err)
+	}
 
-		var (
-			isTooYoung        = false
-			geoResetDifferent = false
-		)
-		for _, v := range cohorts {
-			if isTooYoung = (v == reputation.CohortTooYoung); isTooYoung {
-				break
-			}
-			if geoResetDifferent = (v == reputation.CohortGeoResetDifferent); geoResetDifferent {
-				break
-			}
-		}
+	repClient, ok := ctx.Value(appctx.ReputationClientCTXKey).(reputation.Client)
+	if !ok {
+		return ErrNoReputationClient
+	}
```

Comment:
Nit: I'd think that parsing an uuid would be more expensive than obtaining something from context and type-casting it. In the future, we could probably take that into account.

Reply:
I did indeed deliberate on this and in the end went with this order of statements opposed to having the retrieval first because -
* I wanted to keep the rep client code [here](https://github.com/brave-intl/bat-go/blob/48d76c51f817387a01baae2d14edac5f40ce9970/services/wallet/datastore.go#L562) and [here](https://github.com/brave-intl/bat-go/blob/48d76c51f817387a01baae2d14edac5f40ce9970/services/wallet/datastore.go#L569) close together as they are going to be pulled out and it felt more intuitive
* If the rep client was missing the application would have failed to start so at this point the only unknown is the `walletID` i.e if we did the retrieval and type assertion on the rep client and the `walletID` failed then we wasted the rep client retrieval which we were guaranteed

That was the thought behind the statement order which seemed to make sense at the time :laughing: 

If it wasn't for the above the retrieval would have most likely been first. Hopefully this code will be fixed soon and the client removed from the db layer and context entirely

Reply:
The reasoning makes sense to me, thank you!